src/core/rtos/Threadx.cpp

```diff
@@ -0,0 +1,60 @@
+
+#include <core/rtos/BytePool.hpp>
+#include <core/rtos/Threadx.hpp>
+#include <core/utils/log.hpp>
+
+namespace log = core::log;
+
+namespace core::rtos {
+
+namespace {
+Initializable** initializableList;
+std::size_t initListLength;
+BytePoolBase* mainThreadPool;
+} // namespace
+
+TXError init(Initializable* initList[], std::size_t length, BytePoolBase& pool) {
+    TXError errorCode = pool.init();
+    if (errorCode != SUCCESS)
+        return errorCode;
+
+    return bulkInitialize(initList, length, pool);
+}
+
+extern "C" void tx_application_define(void* first_unused_memory) {
+    init(initializableList, initListLength, *mainThreadPool);
+}
+
+TXError startKernel(Initializable* initList[], std::size_t length, BytePoolBase& pool) {
+    initializableList = initList;
+    initListLength    = length;
+    mainThreadPool    = &pool;
+    tx_kernel_enter();
+    return SUCCESS;
```

Comment:
So tx_kernel_enter calls tx_application_define(), which initializes everything in init_list. Added a comment explaining that. (that is also why we need that anonymous namespace with initlist and everything else in it, because tx_application_define can't take in any user-defined arguments).